src/xnnpack/simd/s32-avx2.h

```diff
@@ -43,6 +43,44 @@ static XNN_INLINE xnn_simd_s32_t xnn_min_s32(xnn_simd_s32_t a,
   return _mm256_min_epi32(a, b);
 }
 
+static XNN_INLINE xnn_simd_s32_t xnn_clz_s32(xnn_simd_s32_t a) {
+  __m128i lowi = _mm256_extracti128_si256(a, 0);
+  __m128i highi = _mm256_extracti128_si256(a, 1);
+
+  __m256d low = _mm256_cvtepi32_pd(lowi);
+  __m256d high = _mm256_cvtepi32_pd(highi);
+  xnn_simd_s32_t low_a = _mm256_castpd_si256(low);
+  xnn_simd_s32_t high_a = _mm256_castpd_si256(high);
+
+  xnn_simd_s32_t shift_low = _mm256_srli_epi64(low_a, 52) & 0x7FF;
+  xnn_simd_s32_t shift_high = _mm256_srli_epi64(high_a, 52) & 0x7FF;
+
+  xnn_simd_s32_t exponent =
+      _mm256_set_epi32(_mm256_extract_epi32(shift_high, 6),
+                       _mm256_extract_epi32(shift_high, 4),
+                       _mm256_extract_epi32(shift_high, 2),
+                       _mm256_extract_epi32(shift_high, 0),
+                       _mm256_extract_epi32(shift_low, 6),
+                       _mm256_extract_epi32(shift_low, 4),
+                       _mm256_extract_epi32(shift_low, 2),
+                       _mm256_extract_epi32(shift_low, 0));
```

Comment:
I don't think this is the most efficient way of doing what I think you're trying to do here.